menu/menu_displaylist.c

```diff
@@ -5287,6 +5287,64 @@ static int menu_displaylist_parse_input_description_list(
    return count;
 }
 
+static int menu_displaylist_parse_netplay_mitm_server_list(
+      menu_displaylist_info_t *info, settings_t *settings)
+{
+   unsigned count              = 0;
+#ifdef HAVE_NETWORKING
+   rarch_system_info_t *system = &runloop_state_get_ptr()->system;
+   size_t menu_index           = 0;
+   char entry_label[21];
+
+   unsigned i;
+   unsigned list_len = ARRAY_SIZE(netplay_mitm_server_list);
+
+   entry_label[0] = '\0';
+
+   if (!system || !settings)
+      goto end;
+
+   for (i = 0; i < list_len; i++)
+   {
+
+       /* Add menu entry */
+       if (menu_entries_append_enum(info->list,
+             netplay_mitm_server_list[i].description,
+             entry_label,
+             MENU_ENUM_LABEL_INPUT_DESCRIPTION,
```

Comment:
Wierd things happens if I do this. Perhaps it should be `MENU_ENUM_LABEL_DEFERRED_DROPDOWN_BOX_LIST_NETPLAY_MITM_SERVER`?   
![1](https://user-images.githubusercontent.com/5746456/166251344-a0c138eb-0d0d-4230-bd13-2d994f04a8f3.jpg)

Reply:
You are right, it must be an empty enum.

Go to msg_hash.h and create a new one (msg_hash_enums), and name it something like "MENU_ENUM_LABEL_NETPLAY_MITM_SERVER_LOCATION", and use this eum instead of MENU_ENUM_LABEL_NETPLAY_MITM_SERVER.

Make sure to add a description at intl/msg_hash_us.c -> msg_hash_get_help_us_enum to this enum; this way we get this description to show up inside the relay server selection submenu when pressing "select".

Reply:
I've added it in latest commit, but I can't see any difference. Perhaps bcs I'm not using gamepad..

Reply:
No, you need to add just that one enum, literally, not with a macro.

Use MENU_ENUM_LABEL_INPUT_BIND_DEVICE_TYPE as an example for both msg_hash.h and msg_hash_us.c.

Reply:
How about now?

Reply:
Looks good; just clean up the for loop at menu_displaylist.c (remove empty lines, comment about input being mapped, etc) and we should be done.

Did you test that all 5 options write the correct MITM handle to the config when set from the submenu?

Reply:
>Looks good; just clean up the for loop at menu_displaylist.c (remove empty lines, comment about input being mapped, etc) and we should be done.

Should be fixed now.

> Did you test that all 5 options write the correct MITM handle to the config when set from the submenu?

yes.

Reply:
Ok, you can mark this as resolved too.